### PR TITLE
Add documentation for Kueue visibility on-demand

### PR DESCRIPTION
**Pull request by PBundyra**
#### What type of PR is this?
/kind documentation

#### What this PR does / why we need it:
This PR continues the development of [KEP#168-2](https://github.com/kubernetes-sigs/kueue/pull/1300)

#### Does this PR introduce a user-facing change?

```release-note
Add documentation for Kueue visibility on-demand

```

### CODE REVIEW

File: site/content/en/docs/tasks/monitor_pending_workloads/_index.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+title: "Monitor pending Workloads"
+linkTitle: "Monitor pending Workloads"
+weight: TBD
+date: TBD
+description: >
+  How to monitor pending Workloads
+no_list: true
+---
+
+Kueue shares two way of monitoring pending Workloads. TBD maybe something about preferred way
```
Comment (tenzen-y) [inline site/content/en/docs/tasks/monitor_pending_workloads/_index.md:12]: > TBD maybe something about preferred way


Is this under working progress?

File: site/content/en/docs/tasks/monitor_pending_workloads/pending_workloads_in_status.md [hunk 2]
```diff
@@ -1,9 +1,9 @@
 ---
-title: "Monitor pending workloads"
+title: "Pending workloads in Status"
 date: 2023-09-27
 weight: 3
 description: >
-  Monitor pending workloads.
+  # Monitor pending workloads.
```
Comment (tenzen-y) [inline site/content/en/docs/tasks/monitor_pending_workloads/pending_workloads_in_status.md:6]: Please update.

File: site/content/en/docs/tasks/monitor_pending_workloads/pending_workloads_on_demand.md [hunk 3]
```diff
@@ -0,0 +1,150 @@
+---
+title: "Pending Workloads on-demand"
+date: 2023-12-04
+weight: 3
+description: >
+#   Monitor pending workloads.
```
Comment (tenzen-y) [inline site/content/en/docs/tasks/monitor_pending_workloads/pending_workloads_on_demand.md:6]: ditto.

File: site/content/en/docs/tasks/monitor_pending_workloads/pending_workloads_on_demand.md [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+
+```shell
+kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/kueue/main/site/static/examples/admin/single-clusterqueue-setup.yaml
+```
+
+To install two local queues, run the following command:
+
+```shell
+kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/kueue/main/site/static/examples/admin/localqueues-setup.yaml
+```
```
Comment (tenzen-y) [inline site/content/en/docs/tasks/monitor_pending_workloads/pending_workloads_on_demand.md:43]: I believe that we should provide the only minimum case. So setting a localqueue by `single-clusterqueue-setup.yaml` should be enough.

Or is there any reason that we should have multiple localqueues?

File: site/content/en/docs/tasks/monitor_pending_workloads/pending_workloads_on_demand.md [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+
+To view pending workloads in ClusterQueue run the following command:
+
+```shell
+kubectl get --raw /apis/visibility.kueue.x-k8s.io/v1alpha1/clusterqueues/cluster-queue/pendingworkloads
+```
+
+You can pass optional query parameters:
+- limit `<int>` - 1000 on default. It indicates max number of pending workloads that should be fetched.
+- offset `<int>` - 0 by default. It indicates position of the first pending workload that should be fetched, starting from 0.
```
Comment (tenzen-y) [inline site/content/en/docs/tasks/monitor_pending_workloads/pending_workloads_on_demand.md:73]:
```suggestion
- limit `<integer>` - 1000 on default. It indicates max number of pending workloads that should be fetched.
- offset `<integer>` - 0 by default. It indicates position of the first pending workload that should be fetched, starting from 0.
```

File: site/content/en/docs/tasks/monitor_pending_workloads/pending_workloads_on_demand.md [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+
+To view only the first 3 pending workloads use:
+```shell
+kubectl get --raw /apis/visibility.kueue.x-k8s.io/v1alpha1/clusterqueues/cluster-queue/pendingworkloads\?limit\=3
```
Comment (tenzen-y) [inline site/content/en/docs/tasks/monitor_pending_workloads/pending_workloads_on_demand.md:77]: ... \?limit\=3

Why do we use the escape character "\\\" here?

File: site/content/en/docs/tasks/monitor_pending_workloads/pending_workloads_on_demand.md [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+
+Similarly to Cluster Queue, to view pending workloads in LocalQueue `team-a-queue` run the following command:
+
+```shell
+kubectl get --raw /apis/visibility.kueue.x-k8s.io/v1alpha1/namespaces/default/localqueues/team-a-queue/pendingworkloads
+```
+
+Similarly to Cluster Queue, you can pass optional query parameters:
+- limit `<int>` - 1000 on default. It indicates max number of pending workloads that should be fetched.
+- offset `<int>` - 0 by default. It indicates position of the first pending workload that should be fetched, starting from 0.
```
Comment (tenzen-y) [inline site/content/en/docs/tasks/monitor_pending_workloads/pending_workloads_on_demand.md:95]:
```suggestion
- limit `<integer>` - 1000 on default. It indicates max number of pending workloads that should be fetched.
- offset `<integer>` - 0 by default. It indicates position of the first pending workload that should be fetched, starting from 0.
```

File: site/content/en/docs/tasks/view_pending_workloads.md
Comment (tenzen-y) [inline site/content/en/docs/tasks/view_pending_workloads.md:1]: What is this? It looks like unrelated to the VisibilityOndemand feature.

File: site/package-lock.json
Comment (tenzen-y) [inline site/package-lock.json:1]: Can you open another PR?

**Review by tenzen-y — COMMENTED** [review]
I left my first comment.

File: site/content/en/docs/tasks/monitor_pending_workloads/_index.md (hunk 1 above)
Comment (mimowo) [reply site/content/en/docs/tasks/monitor_pending_workloads/_index.md:12]: I think in the docs it is safe to promote the on-demand and say it is preferred due to the higher scale supported. It is also preferred because the in-status approach requires additional memory to maintain the snapshot per ClusterQueue.

In the long run we will want to promote on-demand to beta, for in-status, we could probably deprecate and eventually remove. cc @mwielgus

File: site/content/en/docs/tasks/monitor_pending_workloads/pending_workloads_on_demand.md (hunk 4 above)
Comment (mimowo) [reply site/content/en/docs/tasks/monitor_pending_workloads/pending_workloads_on_demand.md:43]: Yeah,  I don't have a strong view, maybe we can go with minimal setup in the first iteration.

One reason to have 2 localqueues is to demonstrate that the positions in ClusterQueue might be interleaving between two LocalQueues, for this purpose we could just install another local queue from a dedicated file.

Comment (PBundyra) [reply site/content/en/docs/tasks/monitor_pending_workloads/pending_workloads_on_demand.md:43]: My intention was to demonstrate two facts:
1) The position of Workloads across LocalQueues is consistent, 
2) A user requesting Workloads belonging to one LocalQueue does not have any information about pending Workloads in other LocalQueues.

Comment (mimowo) [reply site/content/en/docs/tasks/monitor_pending_workloads/pending_workloads_on_demand.md:43]: Sounds reasonable to me, WDYT @tenzen-y ?

File: site/content/en/docs/tasks/monitor_pending_workloads/pending_workloads_on_demand.md (hunk 6 above)
Comment (PBundyra) [reply site/content/en/docs/tasks/monitor_pending_workloads/pending_workloads_on_demand.md:77]: If one doesn't use quotation marks in the command "\" are necessary. I agree that using quotation marks and omitting "\" is a cleaner way so I'll fix that

File: site/content/en/docs/tasks/monitor_pending_workloads/_index.md [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+title: "Monitor pending Workloads"
+linkTitle: "Monitor pending Workloads"
+weight: TBD
+date: TBD
+description: >
+  How to monitor pending Workloads
+no_list: true
+---
+
+Kueue shares two way of monitoring pending Workloads. For Kueue 0.6+ the preferred way to monitor pending Workloads is using the on-demand feature.
```
Comment (alculquicondor) [inline site/content/en/docs/tasks/monitor_pending_workloads/_index.md:12]:
```suggestion
Kueue provides two way of monitoring pending Workloads. For Kueue 0.6 and newer the preferred way to monitor pending Workloads is using the on-demand API.
```

File: site/content/en/docs/tasks/monitor_pending_workloads/_index.md [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+
+- [Pending Workloads in Status](/docs/tasks/monitor_pending_workloads/pending_workloads_in_status).
+- [Pending Workloads on-demand](/docs/tasks/monitor_pending_workloads/pending_workloads_on_demand).
```
Comment (alculquicondor) [inline site/content/en/docs/tasks/monitor_pending_workloads/_index.md:15]: If it's preferred, it should go first.

File: site/content/en/docs/tasks/monitor_pending_workloads/pending_workloads_on_demand.md [hunk 10]
```diff
@@ -0,0 +1,611 @@
...
+
+To install a simple setup of cluster queue
+
+{{< include "examples/admin/single-clusterqueue-setup.yaml" "yaml" >}}
+
+
+run the following command:
+
+```shell
+kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/kueue/main/site/static/examples/admin/single-clusterqueue-setup.yaml
```
Comment (alculquicondor) [inline site/content/en/docs/tasks/monitor_pending_workloads/pending_workloads_on_demand.md:41]: would an URL like this work?
```
kubectl apply --server-side -k "github.com/kubernetes-sigs/kueue/config/default?ref=main"
```
(from our installation guide).

File: site/content/en/docs/tasks/monitor_pending_workloads/pending_workloads_on_demand.md [hunk 11]
```diff
@@ -0,0 +1,611 @@
...
+    }
+  ]
+}
+```
+
+You can pass optional query parameters:
+- limit `<integer>` - 1000 on default. It indicates max number of pending workloads that should be fetched.
+- offset `<integer>` - 0 by default. It indicates position of the first pending workload that should be fetched, starting from 0.
+
+To view only the first 2 pending workloads use:
```
Comment (alculquicondor) [inline site/content/en/docs/tasks/monitor_pending_workloads/pending_workloads_on_demand.md:213]:
```suggestion
For example, to view only the first 2 pending workloads you can run the following command:
```

File: site/content/en/docs/tasks/monitor_pending_workloads/pending_workloads_on_demand.md [hunk 12]
```diff
@@ -0,0 +1,611 @@
...
+      "priority": 0,
+      "localQueueName": "user-queue",
+      "positionInClusterQueue": 1,
+      "positionInLocalQueue": 1
+    }
+  ]
+}
+```
+
+To view pending workloads having at least position 5 in Cluster Queue use:
```
Comment (alculquicondor) [inline site/content/en/docs/tasks/monitor_pending_workloads/pending_workloads_on_demand.md:268]: Too many examples that don't add much value. Mix this within the previous example.

File: site/content/en/docs/tasks/monitor_pending_workloads/pending_workloads_on_demand.md [hunk 13]
```diff
@@ -0,0 +1,611 @@
...
+      "priority": 0,
+      "localQueueName": "user-queue",
+      "positionInClusterQueue": 1,
+      "positionInLocalQueue": 1
+    }
+  ]
+}
+```
+
+To view pending workloads having at least position 5 in Local Queue use:
```
Comment (alculquicondor) [inline site/content/en/docs/tasks/monitor_pending_workloads/pending_workloads_on_demand.md:538]: Same suggestion here.

**Review by alculquicondor — COMMENTED** [review]
/approve

File: site/content/en/docs/tasks/monitor_pending_workloads/pending_workloads_on_demand.md (hunk 10 above)
Comment (PBundyra) [reply site/content/en/docs/tasks/monitor_pending_workloads/pending_workloads_on_demand.md:41]: IIUC this URL installs Kueue on a cluster, and does not create a ClusterQueue so I don't think they are equivalent

Comment (alculquicondor) [reply site/content/en/docs/tasks/monitor_pending_workloads/pending_workloads_on_demand.md:41]: What I mean is the overall shape of the URL (it doesn't have raw). Another alternative would be to point to the website itself, similar to this `https://kueue.sigs.k8s.io/examples/jobs/sample-job.yaml`.

Comment (PBundyra) [reply site/content/en/docs/tasks/monitor_pending_workloads/pending_workloads_on_demand.md]: Oh yes, these URLs work, I've changed it

File: site/content/en/docs/tasks/monitor_pending_workloads/pending_workloads_on_demand.md [hunk 14]
```diff
@@ -0,0 +1,499 @@
...
+
+
+Now, let's create 10 jobs
+
+{{< include "examples/jobs/sample-job.yaml" "yaml" >}}
+
+using a command: 
+
+```shell
+for i in {1..10}; do kubectl create -f https://kueue.sigs.k8s.io/examples/jobs/sample-job.yaml; done
```
Comment (tenzen-y) [inline site/content/en/docs/tasks/monitor_pending_workloads/pending_workloads_on_demand.md:52]:
```suggestion
for i in {1..5}; do kubectl create -f https://kueue.sigs.k8s.io/examples/jobs/sample-job.yaml; done
```

I think that only 5 jobs should be enough since the below response api-server is a bit huge.

**Review by tenzen-y — COMMENTED** [review]
Generally, LGTM

File: site/content/en/docs/tasks/monitor_pending_workloads/pending_workloads_on_demand.md (hunk 14 above)
Comment (tenzen-y) [reply site/content/en/docs/tasks/monitor_pending_workloads/pending_workloads_on_demand.md:52]: Or, we may be able to omit the part of the below responses.

File: site/content/en/docs/tasks/monitor_pending_workloads/pending_workloads_on_demand.md [hunk 15]
```diff
@@ -0,0 +1,295 @@
...
+    }
+  ]
+}
+```
+
+You can pass optional query parameters:
+- limit `<integer>` - 1000 on default. It indicates max number of pending workloads that should be fetched.
+- offset `<integer>` - 0 by default. It indicates position of the first pending workload that should be fetched, starting from 0.
+
+To view only 1 pending workloads use, starting from position 1 in Local Queue run:
```
Comment (mimowo) [inline site/content/en/docs/tasks/monitor_pending_workloads/pending_workloads_on_demand.md:258]:
```suggestion
To view only 1 pending workloads use, starting from position 1 in LocalQueue run:
```
nit, for consistency

File: site/content/en/docs/tasks/monitor_pending_workloads/pending_workloads_on_demand.md [hunk 16]
```diff
@@ -0,0 +1,295 @@
...
+
+### Enabling feature VisibilityOnDemand
+
+VisibilityOnDemand is an `Alpha` feature disabled by default, check the [Change the feature gates configuration](/docs/installation/#change-the-feature-gates-configuration) section of the [Installation](/docs/installation/) for details.
+
+## Monitor pending workloads on demand
+
+> _Available in Kueue v0.6.0 and later_
+
+To install a simple setup of cluster queue
```
Comment (mimowo) [inline site/content/en/docs/tasks/monitor_pending_workloads/pending_workloads_on_demand.md:33]:
```suggestion
To install a simple setup of ClusterQueue
```
ni: for consistency

File: site/content/en/docs/tasks/monitor_pending_workloads/pending_workloads_on_demand.md [hunk 17]
```diff
@@ -0,0 +1,295 @@
...
+    }
+  ]
+}
+```
+
+You can pass optional query parameters:
+- limit `<integer>` - 1000 on default. It indicates max number of pending workloads that should be fetched.
+- offset `<integer>` - 0 by default. It indicates position of the first pending workload that should be fetched, starting from 0.
+
+To view only 1 pending workloads use, starting from position 1 in Cluster Queue run:
```
Comment (mimowo) [inline site/content/en/docs/tasks/monitor_pending_workloads/pending_workloads_on_demand.md:136]:
```suggestion
To view only 1 pending workloads use, starting from position 1 in ClusterQueue run:
```
nit, for consistency

**Review by mimowo — COMMENTED** [review]
LGTM, just nits

File: site/static/examples/visibility/clusterrolebinding-pending-workloads.yaml
[the remark this answers is not in the document]
Comment (tenzen-y) [reply site/static/examples/visibility/clusterrolebinding-pending-workloads.yaml:1]: That makes sense. Thanks!

**Review by tenzen-y — COMMENTED** [review]
Thank you for the updates!
/lgtm
/approve
